.github/workflows/ci.yml

```diff
@@ -25,9 +30,15 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
+      - name: Install Poetry
+        uses: abatilo/actions-poetry@v2.3.0
+        with:
+          poetry-version: 1.8.3
+
```

Comment:
Is setting up Poetry before python the right thing to do in order to use caching?